File: 5.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
  const [a, n] = input.trim().split(' ').map(Number);

   if (a === 1) 
   {
      console.log(n);
   } 
    else if (a === 2) 
    {
        if (n === 1) 
        {
            console.log(-1);
        } 
         else 
         {
             console.log(n * (n - 1));
         }
    } 
     else 
     {
        let eq = 0;
        let dif = n * (n - 1);
         for (let i = 0; i < a - 2; i++) 
         {
             const eq2 = eq;
              eq = dif;
               dif = dif * (n - 2) + eq2 * (n - 1);
         }
          if (dif === 0) 
          {
              console.log(-1);
          } 
           else 
           {
               console.log(dif);
           }
     }

File: 5.cpp
#include <iostream>
using namespace std;

int main() 
{
  int a, n;
   cin >> a >> n;
    if (a == 1) 
    {
       cout << n;
    } 
     else if (a == 2) 
           {
                if (n == 1) 
                {
                    cout << -1;
                } 
                 else 
                 {
                     cout << n * (n - 1);
                 }
            } 
      else 
      {
         long eq = 0, dif = n * (n - 1);
          for (int i = 0; i < a - 2; i++) 
          {
              long eq2 = eq;
               eq = dif;
                dif = dif * (n - 2) + eq2 * (n - 1);
          }
             if (dif == 0) 
             {
                 cout << -1;
             }
              else 
              {
                  cout << dif;
              } 
       }
}
